### ISSUE TITLE
Remove and optimize (hardcoded) LANS

### ISSUE THREAD

**Issue by CaMer0n** [e107inc/e107#6]
Remove all hard-coded terms from e107 and optimize!

Translator's Checklist: 
- Remove any HTML from the LAN (bbcodes are okay)
- Use double-quotes. eg. define("LAN_01", "It's all good"); 
- Make sure "LAN" is used inside the constant name. 
- Do not leave empty or unused defines. eg define("LAN_SOMETHING", "");  instead, comment them out with // 
- Use $tp->lanVars() for variables conversion. eg. [x] etc. 
- Check for duplicate/similar terms (admin only) against lan_admin.php (eg. LAN_UPDATE)
- Check for duplicate/similar terms (frontend or admin) against e107_languages/English/English.php
- Reduce the number of 'defines' as much as possible by using generic terms. 
- Do NOT translate debug information. (eg. anything inside addDebug(); or "E107_DEBUG_LEVEL >0" etc.  ) 
- Do NOT translate anything used within e_DEVELOPER. 
- If you find a recurring word/term appearing in several areas of e107, please mention it here as a 'generic' candidate in this issue.

**Comment by CaMer0n** [comment]
@LaocheXe, @hanklu, @Jimmi08, @Fiedo, @lonalore, @e107bg, @Moc, @nlstart 
This issue is the main priority for beta2. If you wish to see e107 v2 with complete language-packs in other languages, please consider sending a pull-request with English terms inside the code replaced and 'defines' and added to the appropriate language file. Known issues can be found by searching for 'TODO LAN' over the entire project. (If you are unsure of what is required, please let me know and I'll be happy to explain). Thanks.

**Comment by LaocheXe** [comment]
So you want different language files of the TO DO's? So in English.php define("LAN_HELLO1", "Hello"); to German.php define("LAN_HELLO1", "Hallo");?

**Comment by CaMer0n** [comment]
Not quite. Everywhere there is a 'TODO LAN' inside an e107 script (not a language file).. the english needs to be removed from the script and replaced with a new LAN in the corresponding file.

**Comment by CaMer0n** [comment]
Please view the commit above for an example. (I should have removed the 'TODO' comment also)

**Comment by nlstart** [comment]
Use this view on the coding to see which 'TODO' comments are still actual:
https://github.com/e107inc/e107/search?utf8=%E2%9C%93&q=%22todo+lan%22&type=Code

**Comment by LaocheXe** [comment]
Next question: I did the e107_admin>users.php and did the TODO Lans - what do I do with what I got? I don't know much about Github and Pull Request

**Comment by CaMer0n** [comment]
Updates the description above with a checklist.

**Comment by tgtje** [comment]
Question : translation of hidden ALERT texts like:  "A new update is ready to install! Click to unzip and install "

Is there a method to envoke this behaviour so it displays to check if the LAN translations and LAN coding is correct.

Although (mind my php skills :(  ) my editor seems to display the coding appropiate so it LOOKS usable, and updating those files and no blank pages appear (lucky), i have no way of knowing if it still is ok BEFORE sending a pull request...
OR : 
just send a pull request so it can be checked ? what is my best way to go ahead please ?

**Comment by CaMer0n** [comment]
Yes, in admin_shortcodes.php you can see that the alert is rendered using addInfo() which is public. 
The corresponding define would look something like this: 

    define("LAN_XXX", "A new update is ready to install! Click to unzip and install  v[x]");
    $text = e107::getParser()->lanVars('LAN_XXX', $cacheData); 

Test as much as possible in the browser, but it's true that some parts will be difficult to 'reach'.

**Comment by tgtje** [comment]
Thank you, will do

**Comment by tgtje** [comment]
as mentioned : recurring word/term appearing in several areas of e107

the 'common' phrase (txt)            Posted by  
10 different LANs in use for same txt (language/plugins/theme)
candidate for 1 LAN (system wide use) ??

**Comment by CaMer0n** [comment]
Thanks. so let's start a list: 
* click here
* Posted by [x]

**Comment by tgtje** [comment]
Lets start : all here are found in at least 2 DIFFerent LANs (list is still a work in progress, thought i share already Base is text search and compared to other LANs containing the exact word.. Occurence is language files.plugins/themes etc.. (based on master dd about 5 days)


Common term STANDALONE !!! used as definition

Error
Warning
Edit
Delete
Go to page
Go
none
None
Enter code
Search
Name
Cancel
Date
"Date posted"
"IP Address"
"Author"
"Category"
"Guest"
"Next"
"Previous"
"Block"
"Unblock"
"Delete"
"Info"
"Main site administrator"
"Comment"
"Comment"
"User"
"Username: "
"Comments"
"News"
"News"
"Poll"
"Poll"
"Replying to: "
"Article"
"Review"
"Content"
"Download"
"Registered member"
"Moderate comments"
"here"
"Subject"
"Re:"
"Rating"
Contact Us
"Send"
"Incorrect code entered"
some DATE defines
"Send Email"
"Email sent"
"That doesn't appear to be a valid email address"
"Password Reset"
"Submit"
 "PASS" /Pass
"Installed"
"Not Installed"
"Misc"  / misc
"Downloads"
"Members"
"Submit News"
"Integrity Check"
Remember Me
Log In
"User Logged In"
"User Logged Out"
"Guests: "
"Members: "
"On this page: "
"Online"
"viewing"
"on"
"Replying to"
"Forum"
"Thread"
 "Page"        /page
"Print"
"Searching"
"Site Stats"
"Uploads"
"User Settings"
"List New Items"
"User Posts"
"Admin Area"
"Bugtracker"
"faq"       /Faq
"Incorrect password"
"Password"
"Page List"
"Forums"
"by"
"Posted by"
"News Item: "
"Vote"    /vote
"Votes"    /votes
"Not rated"
"All categories"   /All Cat..
"Searching"
"Search For:"
"Check All"
"Uncheck All"
"One day"

note ONLINE_EL5", "Members"  >< identical in same file as ONLINE_EL2 

i will follow up with more results

**Comment by tgtje** [comment]
Still working but maybe this will be more easier to oversee the present LAN NAMES..

“Type”	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_users.php
	define('USRLAN_207', 'Type');
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload.php
	define('LAN_UL_023', 'Type');
	
	E:\e107-master (14)\e107-master\e107_plugins\alt_auth\languages\English\admin_alt_auth.php
	define('LAN_ALT_64', 'Type');
	
"Your Name: "	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload.php
	define("LAN_61", "Your Name: ");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_61", "Your Name: ");
	
"Version"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_footer.php
	define("FOOTLAN_3", "Version");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_language.php
	define("LANG_LAN_109", "Version");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_plugin.php
	define("EPL_ADLAN_11", "Version");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_theme.php
	define("TPVLAN_11", "Version");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_upload.php
	define("UPLLAN_9", "Version");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload.php
	define("LAN_410", "Version");
	
"File"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_admin.php
	define("LAN_FILE", "File");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_lancheck.php
	define("LAN_CHECK_26", "File");
	//define("LAN_CHECK_20", "File");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_newspost.php
	// define("NWSLAN_68", "File");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_plugin.php
	define ('EPL_ADLAN_128',"File");
	define ('EPL_ADLAN_205',"File");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_upload.php
	define("UPLLAN_10", "File");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload.php
	define("LAN_411", "File");
	
	E:\e107-master (14)\e107-master\e107_plugins\download\languages\English\English_admin.php
	define("DOWLAN_13",  "File");
	
"Screenshot"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_upload.php
	define("UPLLAN_12", "Screenshot");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload.php
	define("LAN_412", "Screenshot");
	
"Description"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_admin.php
	define("LAN_DESCRIPTION", "Description");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_plugin.php
	define("EPL_ADLAN_14", "Description");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_upload.php
	define("UPLLAN_13", "Description");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload.php
	define("LAN_413", "Description");
	
	E:\e107-master (14)\e107-master\e107_plugins\download\languages\English\English_admin.php
	define("DOWLAN_18",  "Description"); //FIXME Use Generic
	
	E:\e107-master (14)\e107-master\e107_plugins\download\languages\English\English_front.php
	define("LAN_dl_7", "Description");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_admin.php
	//define("FORLAN_32", "Description");
	
	E:\e107-master (14)\e107-master\e107_plugins\newsfeed\languages\English_admin.php
	//define("NFLAN_36", "Description");
	
"Category"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_newspost.php
	define("NWSLAN_6", "Category");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_users_extended.php
	define("EXTLAN_44", "Category");
	
	E:\e107-master (14)\e107-master\e107_languages\English\English.php
	define("LAN_CATEGORY", "Category");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_submitnews.php
	define("NWSLAN_6", "Category");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload.php
	define("DOWLAN_11", "Category");
	
	E:\e107-master (14)\e107-master\e107_plugins\download\languages\English\English_admin.php
	define("DOWLAN_11",  "Category"); //FIXME Use Generic
	
	E:\e107-master (14)\e107-master\e107_plugins\download\languages\English\English_front.php
	define("LAN_dl_19", "Category"); //FIXME Use Generic
	
"Top Posters"	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_8", "Top Posters");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_FORUM_0010", "Top Posters"); // LAN_429
	define("LAN_429", "Top Posters");
	
"Most Active Threads"	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_7", "Most Active Threads");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_430", "Most Active Threads");
	
"Threads"	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_6", "Threads");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_userposts.php
	define("UP_LAN_6", "Threads");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_47", "Threads");
	
	E:\e107-master (14)\e107-master\e107_plugins\newforumposts_main\languages\English.php
	define("NFPM_LAN_6", "Threads");
	
"Lastpost"	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_5", "Lastpost");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_userposts.php
	define("UP_LAN_5", "Lastpost");
	
	E:\e107-master (14)\e107-master\e107_plugins\newforumposts_main\languages\English.php
	define("NFPM_LAN_5", "Lastpost");
	
"Replies"	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_4", "Replies");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_userposts.php
	define("UP_LAN_4", "Replies");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_FORUM_0003", "Replies"); // LAN_48 / LAN_55
	define("LAN_55", "Replies");
	define("LAN_48", "Replies");
	
	E:\e107-master (14)\e107-master\e107_plugins\newforumposts_main\languages\English.php
	define("NFPM_LAN_4", "Replies");
	
"Views"	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_3", "Views");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_userposts.php
	define("UP_LAN_3", "Views");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_FORUM_1005", "Views"); // LAN_56
	define("LAN_56", "Views");
	
	E:\e107-master (14)\e107-master\e107_plugins\newforumposts_main\languages\English.php
	define("NFPM_LAN_3", "Views");
	
"Poster"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_upload.php
	define("UPLLAN_5", "Poster");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_2", "Poster");
	
	E:\e107-master (14)\e107-master\e107_plugins\newforumposts_main\languages\English.php
	define("NFPM_LAN_2", "Poster");
	
"Thread"	E:\e107-master (14)\e107-master\e107_languages\English\lan_online.php
	define("ONLINE_EL14", "Thread");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("LAN_1", "Thread");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_userposts.php
	define("UP_LAN_2", "Thread");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_53", "Thread");
	
	E:\e107-master (14)\e107-master\e107_plugins\newforumposts_main\languages\English.php
	define("NFPM_LAN_1", "Thread");
	
"Comments"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_admin.php
	define("ADLAN_114", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_admin_log.php
	define("RL_LAN_099", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_newspost.php
	define("NWSLAN_15", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_search.php
	//define("SEALAN_6", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_comment.php
	define("COMLAN_99", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_online.php
	define("COMMENT", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_search.php
	define("LAN_SEARCH_99", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("TOP_LAN_4", "Comments");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_userposts.php
	define("UP_LAN_13", "Comments");
	
	
	E:\e107-master (14)\e107-master\e107_plugins\poll\languages\English.php
	define("POLLAN_27", "Comments");
	
	E:\e107-master (14)\e107-master\e107_plugins\poll\languages\English_admin_poll.php
	define("POLLAN_27", "Comments");
	
	E:\e107-master (14)\e107-master\e107_plugins\rss_menu\languages\English_admin_rss_menu.php
	define("RSS_COM", "Comments");
	define("RSS_PLUGIN_LAN_14", "Comments");
	
"Posts"	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("TOP_LAN_2", "Posts");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_FORUM_2032", "Posts"); // LAN_67
	define("LAN_67", "Posts");
	
"User Name"	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_users.php
	define("USRLAN_78", "User Name");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_top.php
	define("TOP_LAN_1", "User Name");

**Comment by CaMer0n** [comment]
Excellent, I have added some of these to the 'global' English.php file.

**Comment by tgtje** [comment]
follow up (keeps continuing):

"Error"
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_db.php
	define("DBLAN_32", "Error");
	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_users.php
	// define("USFLAN_2", "Error");
	
	E:\e107-master (14)\e107-master\e107_languages\English\English.php
	define("EMESSLAN_TITLE_ERROR", "Error");
	define("LAN_ERROR", "Error");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_email.php
	define("LAN_EMAIL_12", "Error");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_error.php
	define("PAGE_NAME", "Error");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_installer.php
	define("LANINS_011", "Error");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_upload_handler.php
	define("LANUPLOAD_12", "Error");
	
	E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
	define("LAN_20", "Error");
	
	E:\e107-master (14)\e107-master\e107_themes\_blank\languages\English.php
	define("EMESSLAN_TITLE_ERROR", "Error");
	
"Warning"	
	E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_mailout.php
	define("LAN_MAILOUT_42", "Warning");
	
	E:\e107-master (14)\e107-master\e107_languages\English\English.php
	define("EMESSLAN_TITLE_WARNING", "Warning");
	
	E:\e107-master (14)\e107-master\e107_themes\_blank\languages\English.php
	define("EMESSLAN_TITLE_WARNING", "Warning");
	
	
"Delete"	E:\e107-master (14)\e107-master\e107_languages\English\English.php
	define("LAN_DELETE","Delete");
	
	E:\e107-master (14)\e107-master\e107_languages\English\lan_comment.php
	define("COMLAN_3", "Delete");
	
	E:\e107-master (14)\e107-master\e107_plugins\pm\languages\English.php
	define('LAN_PM_52', "Delete");
	
	
"Go to page"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_GOPAGE", "Go to page");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("LAN_02", "Go to page");
	
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_GO", "Go");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("LAN_03", "Go");
	define("LAN_387", "Go");
	
	
"None"	E:\e107-master (16)\e107_languages\English\admin\lan_fileinspector.php
	define("FC_LAN_12", "None");
	define("FR_LAN_21", "none");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_mailout.php
	define("LAN_MAILOUT_96", "none");
	define("LAN_MAILOUT_232", "None");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_users_extended.php
	define("EXTLAN_61", "None");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_NONE", "None");
	
	E:\e107-master (16)\e107_languages\English\lan_user.php
	define("LAN_407", "none");
	
	E:\e107-master (16)\e107_plugins\download\languages\English\English_admin.php
	define("DOWLAN_28",  "None");
	
	
"Anonymous"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_ANONYMOUS", "Anonymous");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	// define("LAN_FORUM_3004", "Anonymous"); // LAN_311
	define("LAN_311", "Anonymous");
	
	
"Yes"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_YES", "Yes");
	
	E:\e107-master (16)\e107_languages\English\lan_signup.php
	define("LAN_201", "Yes");
	//define("LAN_SIGNUP_10", "Yes");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("POLL_507", "yes");
	
	E:\e107-master (16)\e107_plugins\poll\languages\English.php
	define("POLLAN_10", "yes");
	
	E:\e107-master (16)\e107_plugins\poll\languages\English_admin_poll.php
	define("POLLAN_10", "yes");
	
	E:\e107-master (16)\e107_plugins\tagwords\languages\English.php
	define("LAN_TAG_OPT_10", "yes");
	
	
"No"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_NO", "No");
	
	E:\e107-master (16)\e107_languages\English\lan_signup.php
	define("LAN_200", "No");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("POLL_508", "no");
	
	E:\e107-master (16)\e107_plugins\login_menu\languages\English.php
	define("LAN_LOGINMENU_26", "no");
	
	E:\e107-master (16)\e107_plugins\poll\languages\English.php
	define("POLLAN_11", "no");
	
	E:\e107-master (16)\e107_plugins\poll\languages\English_admin_poll.php
	define("POLLAN_11", "no");
	
	E:\e107-master (16)\e107_plugins\tagwords\languages\English.php
	define("LAN_TAG_OPT_11", "no");
	
	
"OK"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_OK", "OK");
	
	E:\e107-master (16)\e107_plugins\pm\languages\English.php
	define("LAN_PM_110", "ok");
	
	
"Continue"	E:\e107-master (16)\e107_languages\English\admin\lan_administrator.php
	define("ADMSLAN_4", "Continue");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_CONTINUE", "Continue");
	
	E:\e107-master (16)\e107_languages\English\lan_installer.php
	define("LANINS_035", "Continue");
	
	E:\e107-master (16)\e107_languages\English\lan_signup.php
	define("LAN_399", "Continue");
	
	
"Enter code"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_ENTER_CODE", "Enter code");
	
	E:\e107-master (16)\e107_languages\English\lan_contact.php
	define("LANCONTACT_16", "Enter Code");
	
	E:\e107-master (16)\e107_languages\English\lan_fpw.php
	define("LAN_FPW2","Enter code");
	
	
"Search"	E:\e107-master (16)\e107_languages\English\admin\lan_eurl.php
	define("LAN_EURL_CORE_SEARCH", "Search");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_SEARCH", "Search");
	
	E:\e107-master (16)\e107_languages\English\lan_online.php
	define("SEARCH", "Searching");
	
	E:\e107-master (16)\e107_languages\English\lan_search.php
	define("PAGE_NAME", "Search");
	define("LAN_180", "Search");
	
	E:\e107-master (16)\e107_languages\English\lan_userposts.php
	define("UP_LAN_12", "Search");
	
	E:\e107-master (16)\e107_languages\English\lan_user_select.php
	define("US_LAN_6", "Search");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	//define("LAN_FORUM_0042", "Search"); /// LAN_180 => Generic term, moved to e107_languages/English.php LAN_SEARCH
	define("LAN_180", "Search");
	
	E:\e107-master (16)\e107_plugins\search_menu\languages\English.php
	define("LAN_180", "Search");
	
	E:\e107-master (16)\e107_plugins\tagwords\languages\English.php
	define("LAN_TAG_SEARCH_2", "Search");
	
	
"Name"	E:\e107-master (16)\e107_languages\English\admin\lan_language.php
	define("LANG_LAN_108", "Name");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_plugin.php
	define("EPL_ADLAN_10", "Name");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_theme.php
	define("TPVLAN_52", "Name");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_updateadmin.php
	define("UDALAN_4", "Name");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_upload.php
	define("UPLLAN_23", "Name");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_users_extended.php
	define("EXTLAN_1", "Name");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_NAME", "Name");
	
	E:\e107-master (16)\e107_languages\English\lan_user_extended.php
	define("UE_LAN_21", "Name");
	
	E:\e107-master (16)\e107_plugins\download\languages\English\English_admin.php
	define("DOWLAN_12",  "Name"); //FIXME Use Generic
	
	E:\e107-master (16)\e107_plugins\download\languages\English\English_front.php
	define("LAN_dl_28", "Name"); //FIXME Use Generic
	
	E:\e107-master (16)\e107_plugins\gsitemap\languages\English_front.php
	define("GSLAN_4", "Name");
	define("GSLAN_25", "Name");
	
	
"Cancel"	E:\e107-master (16)\e107_languages\English\admin\lan_cron.php
	define("LAN_CRON_1", "Name");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_mailout.php
	define("LAN_MAILOUT_38", "Cancel");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_CANCEL","Cancel");
	
	
"Date"	E:\e107-master (16)\e107_languages\English\admin\lan_mailout.php
	define("LAN_MAILOUT_80", "Date");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_plugin.php
	define ('EPL_ADLAN_185',"Date");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_DATE","Date");
	
	E:\e107-master (16)\e107_languages\English\lan_search.php
	define("LAN_SEARCH_68", "Date");
	
	E:\e107-master (16)\e107_languages\English\lan_user_extended.php
	define("UE_LAN_7", "Date");
	
	E:\e107-master (16)\e107_plugins\download\languages\English\English_front.php
	define("LAN_dl_22", "Date"); //FIXME Use Generic
	
	E:\e107-master (16)\e107_plugins\list_new\languages\English_admin_list_new.php
	define("LIST_ADMIN_SECT_13", "date");
	define("LIST_ADMIN_LAN_14", "date");
	
	
"Date posted"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_DATE_POSTED", "Date posted");
	
	E:\e107-master (16)\e107_languages\English\lan_search.php
	define("LAN_SEARCH_50", "Date posted");
	
	
"IP Address"	E:\e107-master (16)\e107_languages\English\admin\lan_banlist.php
	define("BANLAN_84", "IP Address");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_IP", "IP Address");
	
	E:\e107-master (16)\e107_languages\English\lan_userposts.php
	define("UP_LAN_16", "IP Address");
	
	E:\e107-master (16)\e107_plugins\poll\languages\English.php
	define("POLLAN_18", "IP address");
	define("LAN_FORUM_3036", "IP address"); // POLLAN_18
	
	E:\e107-master (16)\e107_plugins\poll\languages\English_admin_poll.php
	define("POLLAN_18", "IP address");
	
	
"Author"	E:\e107-master (16)\e107_languages\English\admin\lan_image.php
	define('IMALAN_117', "Author"); // use LAN_AUTHOR
	
	E:\e107-master (16)\e107_languages\English\admin\lan_language.php
	define("LANG_LAN_110", "Author");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_mailout.php
	define("LAN_MAILOUT_50", "Author");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_plugin.php
	define("EPL_ADLAN_12", "Author");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_theme.php
	define("TPVLAN_4", "Author");
	
	E:\e107-master (16)\e107_languages\English\English.php
	"define(""LAN_AUTHOR"",	""Author"");"
	
	E:\e107-master (16)\e107_plugins\download\languages\English\English_admin.php
	define("DOWLAN_15",  "Author"); //FIXME Use Generic
	
	E:\e107-master (16)\e107_plugins\download\languages\English\English_front.php
	define("LAN_dl_24", "Author"); //FIXME Use Generic
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("LAN_402", "Author");
	
	E:\e107-master (16)\e107_plugins\list_new\languages\English_admin_list_new.php
	define("LIST_ADMIN_SECT_7", "author");
	
	
"Guest"	E:\e107-master (16)\e107_languages\English\admin\lan_modcomment.php
	define("MDCLAN_4", "Guest");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_GUEST", "Guest");
	
	E:\e107-master (16)\e107_languages\English\lan_comment.php
	define("COMLAN_194", "Guest");
	
	E:\e107-master (16)\e107_plugins\download\languages\English\English_front.php
	define("LAN_dl_52", "Guest");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("LAN_FORUM_0063", "guest"); // LAN_418 / LAN_408 (vf)
	define("LAN_194", "Guest");
	define("LAN_418", "guest");
	
	
"Next"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_NEXT", "Next");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("LAN_05", "Next");
	
	
"Previous"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_PREVIOUS", "Previous");
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("LAN_04", "Previous");
	
	
"Login"	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_LOGIN", "Login");
	
	E:\e107-master (16)\e107_languages\English\lan_online.php
	define("LOGIN", "Logging In");   ??????
	
	E:\e107-master (16)\e107_plugins\forum\languages\English\English_front.php
	define("LAN_10", "Login");
	define("LAN_175", "Login");
	
	
"Settings"	E:\e107-master (16)\e107_languages\English\admin\lan_admin.php
	// define("ADLAN_CL_1", "Settings");
	define("LAN_UI_PREF_LABEL", "Settings");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_eurl.php
	define("LAN_EURL_MENU_SETTINGS", "Settings");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_plugin.php
	define ('EPL_ADLAN_147',"settings");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_SETTINGS", "Settings");
	
	
"Password"	E:\e107-master (16)\e107_languages\English\admin\lan_updateadmin.php
	define("UDALAN_5", "Password");
	
	E:\e107-master (16)\e107_languages\English\admin\lan_users.php
	define("USRLAN_62", "Password");
	
	E:\e107-master (16)\e107_languages\English\English.php
	define("LAN_PASSWORD", "Password");
	
	E:\e107-master (16)\e107_languages\English\lan_page.php
	define("LAN_PAGE_9", "Password");

**Comment by CaMer0n** [comment]
Thanks. Many of those are already in \e107_languages\English\English.php
Please feel free to send a pull-request with modifications to replace existing LANs with those in English.php. eg. Search/Replace LAN_PAGE_9  throughout e107 and replace with LAN_PASSWORD.

**Comment by tgtje** [comment]
Example ; would this be the thing we are after.?.. ( despite the LAN's 'related' position #being part of  >eg.  email lan would end up being MASTER LAN...) ( i better ask twice before making changes > a lot of testing has to be done if they are being called> thats priority 1)

TXT description >  Error

E:\e107-master (14)\e107-master\e107_languages\English\English.php
define("LAN_ERROR", "Error");  # MASTER LAN

1 Example : 
TODO  >>>  E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_db.php
define("LAN_ERROR", "Error")  // WAS define("DBLAN_32", "Error"); 

ENDS UP AS >>>E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_db.php
define("LAN_ERROR", "Error");

All these would follow same routine : 

E:\e107-master (14)\e107-master\e107_languages\English\admin\lan_db.php
define("LAN_ERROR", "Error")  // WAS define("DBLAN_32", "Error");
 
E:\e107-master (14)\e107-master\e107_languages\English\lan_email.php
define("LAN_ERROR", "Error")  //WAS define("LAN_EMAIL_12", "Error");

E:\e107-master (14)\e107-master\e107_languages\English\lan_error.php
define("LAN_ERROR", "Error") //WAS define("PAGE_NAME", "Error");

E:\e107-master (14)\e107-master\e107_languages\English\lan_installer.php
define("LAN_ERROR", "Error")  //WAS  define("LANINS_011", "Error");

E:\e107-master (14)\e107-master\e107_languages\English\lan_upload_handler.php
define("LAN_ERROR", "Error")  // WAS   define("LANUPLOAD_12", "Error");

E:\e107-master (14)\e107-master\e107_plugins\forum\languages\English\English_front.php
define("LAN_ERROR", "Error")  // WAS define("LAN_20", "Error");

E:\e107-master (14)\e107-master\e107_themes\_blank\languages\English.php
define("LAN_ERROR", "Error")   //WAS  define("EMESSLAN_TITLE_ERROR", "Error");



In this case (Error)  it would reduce LAN Descriptions (use) from 8 to 1 single.
(not taken in consideration the pull request (proposed changes)  from Veselin Totev at this point).

**Comment by CaMer0n** [comment]
No, that's not correct. Good example: You would need to find the usage of DBLAN_32 in the script (NOT the language file) and change it to LAN_ERROR.

**Comment by tgtje** [comment]
You are correct, MY mistake... 

as example :  indeed i would have to replace DBLAN for the MASTER LAN at file level, so that in the (suppose existing language lan in 'plugin'or alike ) define can be removed (becomes obsolete).

**Comment by Moc** [comment]
@tgtje 
The e107_languages/English/English.php file is the main language file for the frontend area of your website. This file is loaded on every pageload and can be used by any plugin or theme. It is always included. 

In the example of LAN_ERROR. You can simply comment out the duplicate in each language file. They can be removed at at a later stage, for now it's easy to use as a historical reference. 

So, for example, in e107_plugins/forum/languages/English/English_front.php, comment out the duplicate LAN define (which is already present in the English.php LAN file. 
```php
//define("LAN_20", "Error");
```

Then replace `LAN_20` with `LAN_ERROR` everywhere within the forum plugin (frontend). 

Does that make sense?

**Comment by tgtje** [comment]
Yes it does. I mentioned it for these cases.  (decomment or remove) While working at it it is an easy task to decomment the affiliate language file. Better save than sorry ;-) .

While at it : forum language (front php) carries LAN_02 two times.. !!! each with own description... those are tricky.....

**Comment by CaMer0n** [comment]
There are 2 important language tools in the core (developer mode must be enabled in admin -> preferences) which should be used when optimizing LANs. 

* Admin -> Languages -> Language Tools -> Search for Deprecated Lans  (Core language files)
* Plugin Manager -> Plugin Builder -> Check language files (Plugin language files)

When using the core-language file tool, it is VERY important to search EVERY file of e107 before commenting out a language term. Because sometimes multiple language files are loaded by a single script. 

Both tools take load the generic LANs so one can easily see what should be replaced. 
There are some exceptions which occur, particularly with e_search.php and lan_search.php terms.

**Comment by nlstart** [comment]
Fixing the double-quotes has some unintended side effects by search-and-replace of all instances.
See e.g. e107/e107_languages/English/lan_signup.php
line 23: //define("LAN_106", "That doesn"t appear to be a valid email address");
line 73: //define("LAN_SIGNUP_32", "What"s an XUP file?");

**Comment by CaMer0n** [comment]
@nlstart - I checked each file one-by-one. There shouldn't be any errors. Those definitions are already commented out.

**Comment by tgtje** [comment]
Well, instead of a new issue : problem i can not solve (tested..no go)
Case : no existing lan's (or none usable) terms used in english example : 

txt >  Drag files here.  (upload media screen)
eg used in jquery plupload js etc..

txt >  SOME Captions eg. Date/Time / Latest Comments
some are positioned in default install xml..  (trying/using lan defs. have no result)

Articles as example can be altered using chapter template, but how about such ones as above ?

**Comment by tgtje** [comment]
Question for Coding (if possible): method to 'override' ( by symbol or any other means) the verify proces IFF a txt string is identical to the english one eg . term : in  (english = in  dutch = in german =in latin = in ....)
Override best should eliminate the 'comment' on verify : LAN...Identical string (warning only) 
In case more translators are working on something together (as indication the file 'might' need to be translated...
![lans](https://cloud.githubusercontent.com/assets/1811857/8634391/342bec08-27f7-11e5-8f6f-c8b466b75893.jpg)

**Comment by tgtje** [comment]
Question (long story in shortest as can be  @ confusing)
Using deprecated lans function > missing in file ( red exclamation > unused in file)
Do i have to (for sake of check ?) add that particular mentioned (actual existing lan in other file) one, also into the checked against file..? (not deprecated, but displays as warning, not telling it IS needed or NOT). 

(adding = check then ok; for all else it is only adding lines to files ( in english and other languages too...,results in same def in multiple files).

**Comment by tgtje** [comment]
@CaMer0n  Question : Translation >  _Download_ is a CORE PLUGIN (likely always will be); Since several defines have a 'general' textual explanation, would it be agreeable to replace them with general LAN's in _ English php_ ,  so these ** new lan's** can be used site-wide (also in other plugins and files ?)
Example : Timestamp (as title) becomes LAN_TIMESTAMP  (in English php) which can be used in infopanel (title) and in download replacing dowlan_182  (having same 'wording'. Same scenario for IP, Help etc.... ? ( !!  CORE plugins etc ONLY,, and in point in time developers can 'call' these general lan's for use in their product ? )

**Comment by Moc** [comment]
@tgtje Yes. That is exactly what is meant by "Check for duplicate/similar terms (frontend or admin) against e107_languages/English/English.php" in the first post in this issue. Most of these are already defined in English and need to be applied to the plugins (core or non-core - for plugin developers).

**Comment by CaMer0n** [comment]
@tgtje Just to note that LAN candidates for the English.php file should appear at least 3 (preferably 4) or more times throughout the core. (including plugins) and be used in the frontend also.. 

ie. The term "Timestamp" appears only twice, and both times only in the admin area - so it is **not** a candidate (used less than 3 times) as a generic LAN, but if it were, it would be for  `English/admin/lan_admin.php` (Generic Terms for Admin) not `English/English.php` (Generic terms for frontend + admin).

**Comment by tgtje** [comment]
Ok sirs, thats agreed.. 3 > pref more before general lan usage
As for Timestamp > was idea for lan admin, but found the same in downlan hence the question arose

**Comment by MikeyGMT** [comment]
Is the correct way to replace lans to remove rather than comment out deprecated lines?
For example, rather than changing 
define("ADLAN_4", "Preferences");
to 
// define("ADLAN_4", "Preferences");//LAN_PREFERENCES
Can I suggest we remove the redundant line?
It would help reduce the file size. 

Adding New Lans ... 
can we refrain from re-using LAN_123, when replaced with a generic LAN?
can we also instead of using LAN_123 use LAN_ABBREVIATED_MSG?
This would help make code more readable, and make the task of translating cleaner.

I'm thinking that if LAN_123 is in the code somewhere, but not in the LAN files then the verifier could add a new check to show that there is a redundant LAN in a language pack, and therefore can be removed and not need translation. 

Thanks

Mike

**Comment by CaMer0n** [comment]
@MikeyGMT I'm not sure what the purpose of renaming ADLAN_4 to LAN_PREFERENCES is? Is it a real-life example? It looks nicer, but it will break any old plugin that happened to be using ADLAN_4 if you comment it out. And if you leave it, then we have duplication. So if it's already in lan_admin.php that is enough. 

Generic LANs don't use numbers, while all others should. Hope that answers the question.

**Comment by MikeyGMT** [comment]
@CaMer0n a couple of aims :-
* Reduce translation from 4k+ defines / ease the translation task (yes some are commented out in that count)
* Cleaner and easier to read (and learn) code, e.g. LAN_SOMETHING is easier to read and remember than LAN_1234 means "Something". 
* Removing commented lines, reduce the filesize, improve performance, all be it only a small performance gain.
* GitHub remembers the old versions of files, hence no need to comment. 

You are right though, I'm learning still, and I had not considered non-core plugins. Prefs was just an example picked out as a commonly defined one. For example, ADLAN_4, DBLAN_96, EPL_ADLAN_110, SEALAN_42, WMLAN_08 are all "Preferences", and if you are not fully bi-lingual, then the task of translating is long. Another example, "Preferences Saved" may be consider simply using more generic "Saved". 

I'm aiming to simplify things, rather than break things, hopefully I have not broken too much :)

**Comment by MikeyGMT** [comment]
I've just had a play with Languages > Deprecated LANS, chosen 

admin/lan_updateadmin.php + Script > LAN + Autodetect. 

OK so learned that commented out LANs show up as Disabled, so commenting out is actually useful.

UDALAN_4 "Name" has been identified as Common (Use LAN_NAME instead). 

Interesting.

**Comment by CaMer0n** [comment]
Thanks @MikeyGMT , :+1: for the aims. Commenting-out is indeed useful in case something is broken and we need to find what the original term was quickly. 
The "Deprecated LANs" is very useful to finding issues, but doesn't check includes etc.. so testing after commenting out unused terms is a must. Most of the 'common term' stuff should be okay to implement. (taking into consideration if it's an admin-only term or frontend)

**Comment by rica-carv** [comment]
Line 84 on [banner/e_shortcode.php](https://github.com/rica-carv/e107/blob/ee011f5a212b420dbb556ca6aaa0ff896ca9ff1e/e107_plugins/banner/e_shortcode.php)....

**Comment by MikeyGMT** [comment]
Perhaps an old dilemma, but I want to pose the question:
We could use LAN_SAVE and LAN_SAVED in the front end e.g. in usersettings (+ its shortcode) and in other places.
We are keen to keep English.php small. 
We don't want to create duplication by creating yet another define. 
At the same time we want provide continuity for existing lan packs which have been translated. 
 _How annoying is it that a LAN_DEFINE keeps changing name or moving around? The translator has to keep re-translating the same!_ ... If you see what I mean.

Maybe we could find a middle ground by:
Adding a deserving new define, in the right place, such as LAN_SAVE in English.php, but leave the old define for backward compatibility AND flag the old define as //DUP_LAN_FOR_REMOVAL 

Hence before a Version Release - MultiLan or Core Lan Admin, could have some function to check & list LAN's for removal. When verified, those could be commented out in cleanly.  

I can see that this could get messy and needs some more thought. Hence for now I will be moving LAN_SAVE and LAN_SAVED to English.php but will comment out as normal. (LAN_PREVIEW is another example, it's defined in the lan_admin but is needed in image.php.

Maybe there is a case for English.php to carry only phrases for public visitors, and a new lan which carries action messages for logged in members? I suspect that's a lot of work!

**Comment by MikeyGMT** [comment]
I'm doing some lans work in System Logs area and came across LAN_IP being defined twice and both are slightly differently defined. 

English.php 
define("LAN_IP", "IP Address");

lan_admin.php 
define("LAN_IP","IP");

Does anyone object to commenting the one in admin and correcting the one in English.php the shorter version? i.e. 

define("LAN_IP","IP");

If we really need "IP Address" then we may want to get LAN_IP_ADDRESS created, but I don't see much point. 

Thanks
Mikey

**Comment by Moc** [comment]
> Does anyone object to commenting the one in admin and correcting the one in English.php the shorter version? 
If we really need "IP Address" then we may want to get LAN_IP_ADDRESS created, but I don't see much point.

I'd actually vote for having both LAN_IP and LAN_IP_ADDRESS because I can see use cases for both. In many cases the short version will suffice, but for example in table captions (Admin UI), I would prefer to see 'IP Address' as this in the end is the correct (formal) way of writing the 'concept'. If I'd had to choose between either, my vote is actually for the longer formal one.

**Comment by MikeyGMT** [comment]
OK I'll define both, in English.php then there will be a task to change LAN_IP to LAN_IP_ADDRESS in appropriate places. The short form can be reserved for use where space is an issue.

**Comment by MikeyGMT** [comment]
In Rolling Log, there is a define for the View Log Menu, but I don't see that it is working. 
define("RL_LAN_024", "From"); 
But it looks like in the UI it's now called "Caller" 
"From" is defined in 3 other areas Users, PM, Email, including for the front end. 
I'll add to English.php if no objections but I will use separate pulls - System Logs and then one for the other instances of From.

**Comment by MikeyGMT** [comment]
define("LAN_PLUGIN_DOWNLOAD_NAME", "Plugin Download Name"); 
Is entirely missing from lans, referenced many times in frontend and admin.
Added to English.php as used throughout. 
Tested in admin>prefs>search, rendered correctly. 
Will pull shortly.

**Comment by Moc** [comment]
@MikeyGMT It is defined properly in `/e107_plugins/download/languages/English/English_global.php` where it should be :) 

https://github.com/e107inc/e107/blob/master/e107_plugins/download/languages/English/English_global.php#L3

See: http://e107.org/developer-manual/plugin-basics#languages

**Comment by MikeyGMT** [comment]
Does not render in Settings >Search  

![image](https://cloud.githubusercontent.com/assets/2847651/20031626/6d194950-a372-11e6-8eba-3e1b5c075573.png)

Hence there is some bug. 

Searched for the define in LAN Files - it's used all over the place

Search "lan_plugin_download" (36 hits in 13 files)
  C:\Users\Mikey\Documents\GitHub\e107\e107_handlers\search\comments_download.php (1 hit)
	Line 13: $comments_title = LAN_PLUGIN_DOWNLOAD_NAME;
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\download_shortcodes.php (3 hits)
	Line 49: 				$breadcrumb[]	= array('text' => LAN_PLUGIN_DOWNLOAD_NAME,							'url' => $url->create('download/index'));
	Line 56: 				$breadcrumb[]	= array('text' => LAN_PLUGIN_DOWNLOAD_NAME,							'url' => $url->create('download/index'));
	Line 60: 				$breadcrumb[]	= array('text' => LAN_PLUGIN_DOWNLOAD_NAME,							'url' => $url->create('download/index'));
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\e_frontpage.php (1 hit)
	Line 19: 		$frontPage = array('page' => '{e_PLUGIN}download/download.php', 'title' => LAN_PLUGIN_DOWNLOAD_NAME);
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\e_rss.php (2 hits)
	Line 23: 	$feed['name']		= LAN_PLUGIN_DOWNLOAD_NAME;
	Line 41: 			$feed['name']		= LAN_PLUGIN_DOWNLOAD_NAME.' > '.$rowi['download_category_name'];
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\e_search.php (1 hit)
	Line 40: 			'name'			=> LAN_PLUGIN_DOWNLOAD_NAME,
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\e_upload.php (1 hit)
	Line 22: 			'name'			    => LAN_PLUGIN_DOWNLOAD_NAME, // Prune downloads history
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\handlers\download_class.php (16 hits)
	Line 214: 			return $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, "<div style='text-align:center'>".LAN_NO_RECORDS_FOUND."</div>",'download-categories',true);
	Line 243: 		$caption = varset($DOWNLOAD_CAT_CAPTION) ? $tp->parseTemplate($DOWNLOAD_CAT_CAPTION, TRUE, $sc) : LAN_PLUGIN_DOWNLOAD_NAME;
	Line 327: 			return $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, "<div style='text-align:center'>".LAN_NO_RECORDS_FOUND."</div>", 'download-view', true);
	Line 345: 	    	$DL_VIEW_PAGETITLE = LAN_PLUGIN_DOWNLOAD_NAME." / {DOWNLOAD_CATEGORY} / {DOWNLOAD_VIEW_NAME}";
	Line 474: 	   	   define("e_PAGETITLE", LAN_PLUGIN_DOWNLOAD_NAME." / ".$dlrow['download_category_name']);
	Line 478: 	   	   define("e_PAGETITLE", LAN_PLUGIN_DOWNLOAD_NAME);
	Line 479: 	   	   return $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, "<div class='alert alert-info' style='text-align:center'>".LAN_dl_3."</div>",'download-list',true);
	Line 550: 			$ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, "
	Line 614: 			$text .= $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, $dl_text, 'download-list', true);
	Line 706: 			define("e_PAGETITLE", LAN_PLUGIN_DOWNLOAD_NAME." / ".LAN_dl_47);
	Line 712: 			return $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, $text, 'download-report', true);
	Line 716: 			define("e_PAGETITLE", LAN_PLUGIN_DOWNLOAD_NAME." / ".LAN_dl_51." ".$download_name);
	Line 720: 			$breadcrumb[]	= array('text' => LAN_PLUGIN_DOWNLOAD_NAME,						'url' => e_SELF);
	Line 738: 			return $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, $text, 'download-report', true);
	Line 856: 		   	return $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, $dl_text, 'download-mirror', true);	
	Line 900: 		return $ns->tablerender(LAN_PLUGIN_DOWNLOAD_NAME, $header. "<div class='alert alert-error alert-danger alert-block' style='text-align:center'>".$errmsg."</div>". $footer, 'download-error', true);
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\includes\admin.php (4 hits)
	Line 123: 	protected $menuTitle = LAN_PLUGIN_DOWNLOAD_NAME;
	Line 129: 		protected $pluginTitle	= LAN_PLUGIN_DOWNLOAD_NAME;
	Line 210: 		protected $pluginTitle = LAN_PLUGIN_DOWNLOAD_NAME;
	Line 2034: 		         "download_requested" => LAN_PLUGIN_DOWNLOAD_NAME,
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\languages\English\English_admin.php (1 hit)
	Line 12: // define("LAN_PLUGIN_DOWNLOAD_NAME",	   "Downloads");
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\languages\English\English_global.php (2 hits)
	Line 3: define("LAN_PLUGIN_DOWNLOAD_NAME",  "Downloads");
	Line 4: define("LAN_PLUGIN_DOWNLOAD_DIZ",  "This plugin is a fully featured File-download system");
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\templates\download_template.php (2 hits)
	Line 199: $DL_VIEW_PAGETITLE = LAN_PLUGIN_DOWNLOAD_NAME." / {DOWNLOAD_CATEGORY} / {DOWNLOAD_VIEW_NAME}";
	Line 441: $DOWNLOAD_TEMPLATE['view']['caption'] = LAN_PLUGIN_DOWNLOAD_NAME;
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\url\sef_url.php (1 hit)
	Line 52: 				'name' => LAN_PLUGIN_DOWNLOAD_NAME, // Module name
  C:\Users\Mikey\Documents\GitHub\e107\e107_plugins\download\url\url.php (1 hit)
	Line 111: 				'name'          => LAN_PLUGIN_DOWNLOAD_NAME, // Module name

I've also used the incorrect value for the define, so I will correct that, it's a common term so should be. 
LAN_DOWNLOADS Downloads

Leave it with me.

**Comment by Moc** [comment]
> Does not render in Settings >Search

Some issue on your site. Works fine on all my installations. Please update from github and reinstall the download plugin to test. 

> I've also used the incorrect value for the define, so I will correct that, it's a common term so should be. 
LAN_DOWNLOADS Downloads

Please hold up a minute. 
- LAN_PLUGIN_DOWNLOAD_NAME is a valid LAN reference as I explained in my previous post. The Settings > Search pages contains this reference because it's being hooked into by the Downloads plugin (`e_search.php`) addon. 

- The LAN_DOWNLOAD, defined as "Download" refers to a generic verb "To download". It is not related to the Downloads plugin is therefore defined in the common `/e107_languages/English/English.php` file

Get me on Gitter if you need further explanation.

**Comment by MikeyGMT** [comment]
Try uninstalling downloads ... on two of my sites, display LAN_PLUGIN_DOWNLOAD_NAME as per the screenshot above. Downloads plugin is uninstalled on two of my sites as I don't use. 

If Downloads Plugin is not installed maybe it's a bug in Search Config - i.e. should not display Downloads Setting in Search Config form...?

**Comment by Moc** [comment]
@MikeyGMT 
- In Tools > Database try the `Scan plugin directories` option. 
- See if the LAN reference still occurs after that. It should not display if the download plugin is not installed, you are right about that. 
- After running the plugin directory changes tool, try installing and uninstaling the plugin again. Does it disappear?

If not, please open up a new issue on Github so we can discuss it there.

**Comment by MikeyGMT** [comment]
Scan Plugin Directories shows no errors for Downloads. Installed and uninstalled. Still displayed. Logged issue #2003.

**Comment by MikeyGMT** [comment]
Looking at lans for updateadmin.php mostly duplicates. In fact why not send admin to usersettings? Seems like redundant code?

**Comment by tgtje** [comment]
Question how to proceed wit these : see issue #2659  
Explanation : on https://github.com/e107inc/e107/pull/2232/commits/2c68d9ee52eaf52bb80478420560bd5ef9269df6#diff-d7f2372327954fe64546996f2f909c95 (_closed/merged pull request_) on oldpolls some LAN's were replace by new LAN's  (LAN_FROM & LAN_TO new > where pollan_38; and 37 (being commented out). The other new ones were added and put in admin/lan_admin php...  Now LAN_FROM and LAN_TO  (representing 'from' and 'to' were never incorporated.. Should these LAN"s (pollan 37;38) be reinstated to use (and oldpoll php be recoded OR the new LAN"s be added to the poll/English php (LAN_FROM & TO )and the coding left untouched ?  Please some guidance here or advice how to....

**Comment by Moc** [comment]
@CaMer0n was this one closed intentionally?

**Comment by CaMer0n** [comment]
@Moc No. Looks like it came with the pull-request.
